syntax = "proto2";



service AddressBookService {
    rpc InsertAddress (Person) returns (Response) {};
    rpc DisplayAddresses (Request) returns (AddressBook) {};
    rpc LookUpAddresses (Request) returns (Result) {};
}



message Person {
  required string name = 1;
  required int32 id = 2;
  optional string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phone = 4;
}

message AddressBook {
  repeated Person contacts = 1;
}

message Response {
  optional string message = 1;
}

message Request {
  optional string message = 1;
}

message Result {
  repeated Person results = 1;
}
